File: VB/CustomGridWinApp1/CustomGridControl/MyGridViewInfoRegistrator.vb
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.ViewInfo

Namespace CustomGridControl
	Public Class MyGridViewInfoRegistrator
		Inherits GridInfoRegistrator

		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyGridView"
			End Get
		End Property

		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New MyGridView(TryCast(grid, GridControl))
		End Function

		Public Overrides Function CreateViewInfo(ByVal view As BaseView) As BaseViewInfo
			Return New MyGridViewInfo(TryCast(view, MyGridView))
		End Function
	End Class
End Namespace

File: CS/CustomGridWinApp1/CustomGridControl/MyGridViewInfoRegistrator.cs
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.ViewInfo;

namespace CustomGridControl
{
	public class MyGridViewInfoRegistrator : GridInfoRegistrator
	{
		public override string ViewName
		{
			get
			{
				return "MyGridView";
			}
		}

		public override BaseView CreateView(GridControl grid)
		{
			return new MyGridView(grid as GridControl);
		}

		public override BaseViewInfo CreateViewInfo(BaseView view)
		{
			return new MyGridViewInfo(view as MyGridView);
		}
	}
}
